include/revng/ABIAnalyses/Generated/DeadRegisterArgumentsOfFunction.h

```diff
@@ -0,0 +1,169 @@
+           || (Lh == LatticeElement::NoOrDead && Rh == LatticeElement::Maybe)
+           || (Lh == LatticeElement::NoOrDead && Rh == LatticeElement::Unknown);
+  }
+
+  static LatticeElement
+  combineValues(const LatticeElement &Lh, const LatticeElement &Rh) {
+    if ((Lh == LatticeElement::Maybe && Rh == LatticeElement::NoOrDead)
+        || (Lh == LatticeElement::NoOrDead && Rh == LatticeElement::Maybe)) {
+      return LatticeElement::Maybe;
+    } else if ((Lh == LatticeElement::Maybe && Rh == LatticeElement::Unknown)
+               || (Lh == LatticeElement::NoOrDead
+                   && Rh == LatticeElement::Unknown)
+               || (Lh == LatticeElement::Unknown && Rh == LatticeElement::Maybe)
+               || (Lh == LatticeElement::Unknown
+                   && Rh == LatticeElement::NoOrDead)) {
+      return LatticeElement::Unknown;
+    }
+    return Lh;
+  }
+
+  static LatticeElement transfer(TransferFunction T, const LatticeElement &E) {
+    switch (T) {
+    case TransferFunction::Read:
+      switch (E) {
+      case LatticeElement::Maybe:
+        return LatticeElement::Unknown;
+      case LatticeElement::Unknown:
+        return LatticeElement::Unknown;
+      case LatticeElement::NoOrDead:
+        return LatticeElement::NoOrDead;
+      default:
+        return E;
+      }
+      return E;
+
+    case TransferFunction::WeakWrite:
+      switch (E) {
+      case LatticeElement::Maybe:
+        return LatticeElement::NoOrDead;
+      case LatticeElement::Unknown:
+        return LatticeElement::Unknown;
+      case LatticeElement::NoOrDead:
+        return LatticeElement::NoOrDead;
+      default:
+        return E;
+      }
+      return E;
+
+    case TransferFunction::Write:
+      switch (E) {
+      case LatticeElement::Maybe:
+        return LatticeElement::NoOrDead;
+      case LatticeElement::Unknown:
+        return LatticeElement::Unknown;
+      case LatticeElement::NoOrDead:
+        return LatticeElement::NoOrDead;
+      default:
+        return E;
+      }
+      return E;
+
+    default:
+      return E;
+    }
+  }
+};
+template<bool isForward>
+struct MFI : ABIAnalyses::ABIAnalysis {
+  using LatticeElement = llvm::DenseMap<const llvm::GlobalVariable *,
+                                        CoreLattice::LatticeElement>;
+  using Label = const llvm::BasicBlock *;
+  using GraphType = std::conditional_t<isForward,
+                                       const llvm::BasicBlock *,
+                                       llvm::Inverse<const llvm::BasicBlock *>>;
+  using GT = llvm::GraphTraits<GraphType>;
+  using LGT = GraphType;
+
+  LatticeElement
+  combineValues(const LatticeElement &Lh, const LatticeElement &Rh) const {
+    return ABIAnalyses::combineValues<LatticeElement, CoreLattice>(Lh, Rh);
+  };
+
+  bool isLessOrEqual(const LatticeElement &Lh, const LatticeElement &Rh) const {
+    return ABIAnalyses::isLessOrEqual<LatticeElement, CoreLattice>(Lh, Rh);
+  };
```

Comment:
I think all this stuff is basically a literal in the code generator.
Move this into a base class (using template + CRTP).

Let's make generators generate what's strictly required.

Reply:
moved out of the generated code